src/scripts/main.js

```diff
@@ -1,3 +1,268 @@
 'use strict';
 
-// write code here
+// notification
+const pushNotification = (posTop, posRight, title, type) => {
+  const notification = document.createElement('div');
+  const h2 = document.createElement('h2');
+
+  h2.innerText = title;
+  h2.classList.add('title');
+
+  notification.classList.add('notification', `${type}`);
+  notification.setAttribute('data-qa', 'notification');
+  notification.appendChild(h2);
+
+  const page = document.body;
+
+  page.insertAdjacentElement('afterbegin', notification);
+  notification.style.top = `${posTop}px`;
+  notification.style.right = `${posRight}px`;
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, 2000);
+};
+
+// sorting
+let currentSortColumn = -1;
+let currentSortOrder = 'ASC';
+
+function sortBy(sortByWhat, sortCount) {
+  const list = document.querySelectorAll('tbody tr');
+  const arrOfList = [...list];
+
+  arrOfList.sort((employee1, employee2) => {
+    const employee1Text = employee1.cells[sortCount].textContent;
+    const employee2Text = employee2.cells[sortCount].textContent;
+
+    switch (sortByWhat) {
+      case 'Name':
+      case 'Position':
+      case 'Office':
+        return employee1Text.localeCompare(employee2Text);
+
+      case 'Age':
+        return +employee1Text - +employee2Text;
+
+      case 'Salary':
+        return (
+          +employee1Text.replaceAll(/[$,]/g, '') -
+          +employee2Text.replaceAll(/[$,]/g, '')
+        );
+    }
+  });
+
+  if (currentSortColumn === sortCount) {
+    currentSortOrder = currentSortOrder === 'ASC' ? 'DESC' : 'ASC';
+  } else {
+    currentSortOrder = 'ASC';
+    currentSortColumn = sortCount;
+  }
+
+  if (currentSortOrder === 'DESC') {
+    arrOfList.reverse();
+  }
+
+  for (let i = 0; i < list.length; i++) {
+    list[i].outerHTML = arrOfList[i].outerHTML;
+  }
+}
+
+const thead = document.querySelector('thead tr');
+
+thead.addEventListener('click', (e) => {
+  if (e.target.tagName === 'TH') {
+    sortBy(e.target.textContent, e.target.cellIndex);
+  }
+});
+
+// selected row
+const tbody = document.querySelector('tbody');
+
+tbody.addEventListener('click', (e) => {
+  if (e.target.tagName === 'TD') {
+    if (tbody.querySelector('.active')) {
+      tbody.querySelector('.active').classList.remove('active');
+    }
+    e.target.closest('tr').classList.add('active');
+  }
+});
+
+// fill form
+const table = document.querySelector('table');
+const form = document.createElement('form');
+
+form.classList.add('new-employee-form');
+
+const theaders = ['Name', 'Position', 'Office', 'Age', 'Salary'];
+
+theaders.map((theader) => {
+  const label = document.createElement('label');
+  const input = document.createElement('input');
+
+  label.innerText = `${theader}: `;
+
+  if (theader === 'Office') {
+    const selec = document.createElement('select');
```

Comment:
```suggestion
    const selectInput = document.createElement('select');
```
or selectElement, etc